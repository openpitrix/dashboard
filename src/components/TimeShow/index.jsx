import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';

import { formatTime } from 'utils';

import styles from './index.scss';

export default class TimeShow extends Component {
  static propTypes = {
    time: PropTypes.string,
    type: PropTypes.oneOf(['listTime', 'detailTime', 'lineTime'])
  };

  static defaultProps = {
    type: 'listTime'
  };

  render() {
    const { time, type } = this.props;

    const trans = {
      listTime: 'YYYY/MM/DD',
      detailTime: 'YYYY/MM/DD HH:mm:ss',
      lineTime: 'HH:mm:ss'
    };

    return (
      <Fragment>
        <div className={styles[type]}>{formatTime(time, trans[type])}</div>
      </Fragment>
    );
  }
}
