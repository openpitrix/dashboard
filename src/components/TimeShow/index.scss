@import "~scss/vars";

.detailTime {
  display: inline-block;
  font-size: 12px;
}

.listTime {
  font-size: 12px;
  color: $N100;
}

.lineTime {
  display: inline-block;
  margin-right: 5px;
  font-size: 12px;
  color: $N100;
}
