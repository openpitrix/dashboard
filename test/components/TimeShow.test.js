import TimeShow from 'components/TimeShow';

describe('Component TimeShow', () => {
  it('basic render', () => {
    const wrapper = render(<TimeShow time='2019-04-09T03:39:28Z' />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
