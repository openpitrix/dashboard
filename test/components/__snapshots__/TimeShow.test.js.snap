// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component TimeShow basic render 1`] = `
<div
  class="detailTime"
>
  2019/04/09 11:39:28
</div>
`;
